### PR TITLE
Enrich 'installation path must be an empty directory' errors to help troubleshoot.

### PR DESCRIPTION
**Pull request by mitchell-as**
<table>
<td>
  <a href="https://activestatef.atlassian.net/browse/DX-2881" title="DX-2881" target="_blank">BugDX-2881</a>  Installation path must be an empty directory
  </td></table>
  <br />

### CODE REVIEW

File: cmd/state-installer/cmd.go [hunk 1]
```diff
@@ -296,6 +298,17 @@ func execute(out output.Outputer, cfg *config.Instance, an analytics.Dispatcher,
...
 		if !empty {
+			if usingDefaultInstallPath {
+				// We're having trouble pinning down why these errors are occurring, so report the list of
+				// existing files to Rollbar to help diagnose.
+				if files, err := os.ReadDir(params.path); err == nil {
+					fileList := []string{}
+					for _, file := range files {
+						fileList = append(fileList, filepath.Join(params.path, file.Name()))
+					}
+					rollbar.Critical("Installation path must be an empty directory: %s\nExisting files:\n%s", params.path, strings.Join(fileList, "\n"))
```
Comment (Naatan) [inline cmd/state-installer/cmd.go:309]:
```suggestion
				if files, err := fileutils.ListDirSimple(params.path, false); err == nil {
					rollbar.Critical("Installation path must be an empty directory: %s\nExisting files:\n%s", params.path, strings.Join(files, "\n"))
```

File: cmd/state-installer/installer.go [hunk 2]
```diff
@@ -235,6 +235,10 @@ func installedOnPath(installRoot, channel string) (bool, string, error) {
 		return false, "", nil
 	}
 
+	if !fileutils.FileExists(filepath.Join(installRoot, installation.InstallDirMarker)) {
+		return false, "", nil
+	}
```
Comment (Naatan) [inline cmd/state-installer/installer.go:240]: What I meant with the AC is to return True if this was encountered. So add it to the candidates below.

I think we're avoiding asserting false if it didn't exist in order to support older state tool versions that didn't supply this marker.

**Review by Naatan — CHANGES_REQUESTED** [review]

File: cmd/state-installer/cmd.go (hunk 1 above)
Comment (mitchell-as) [reply cmd/state-installer/cmd.go:309]: `fileutils.ListDirSimple()` does a recursive listing, which this ticket explicitly said not to do. Should I still use your suggestion?

Comment (Naatan) [reply cmd/state-installer/cmd.go:309]: Ahh my bad, no; thanks for spotting that!

**Review by Naatan — APPROVED** [review]

**Pull request merged**